### PR TITLE
nixos/octoprint: add package option

### PR DESCRIPTION
**Pull request by ethancedwards8**
###### Motivation for this change
Add a package option so that you can change what package octoprint uses.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lovesegfault** [comment]
Why do you need this as opposed to just overlaying the octoprint package?

**Comment by ethancedwards8** [comment]
Well, most of the other modules have this option, so standardizing this one is an improvement, IMO. Also, that probably wouldn't occur to the new NixOS user without looking at the code. To iterate on that, I **did** look at the code and it didn't occur to me to do that. Also being able to easily switch out packages on the fly, say you have multiple flake inputs, this was my use case, without having to edit the override is helpful.

File: nixos/modules/services/misc/octoprint.nix [hunk 1]
```diff
@@ -17,9 +17,7 @@ let
 
   cfgUpdate = pkgs.writeText "octoprint-config.yaml" (builtins.toJSON fullConfig);
 
-  pluginsEnv = package.python.withPackages (ps: [ps.octoprint] ++ (cfg.plugins ps));
-
-  package = pkgs.octoprint;
+  pluginsEnv = cfg.package.python.withPackages (ps: [ps.octoprint] ++ (cfg.plugins ps));
```
Comment (jonringer) [inline nixos/modules/services/misc/octoprint.nix:20]: this is probably not the intended affect, as the `package` which gets passed gets replaced by the one which exists in the python3Packages set

Comment (ethancedwards8) [reply nixos/modules/services/misc/octoprint.nix:20]: I see, is there a way around that?

Comment (jonringer) [reply nixos/modules/services/misc/octoprint.nix:20]: Best way would probably to be to create an overlay for the python package set, then build a `python.withPackage` environment with that overlay applied.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-override-a-python-package-how-to-override-a-python-package is what I'm talking about

**Comment by ethancedwards8** [comment]
If someone wants to take over this PR, feel free.

**Comment by ethancedwards8** [comment]
I no longer use this.

**Pull request closed without merging**